### PR TITLE
Alerting: Fetch receivers from remote Alertmanager

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR allows the `externalAlertanager` struct to fetch receivers from the remote Alertmanager.

### CODE REVIEW

File: pkg/services/ngalert/notifier/external_alertmanager.go [hunk 1]
```diff
@@ -267,3 +280,33 @@ func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 
 	return r.next.RoundTrip(req)
 }
+
+// TODO: change implementation, this is only useful for testing other methods.
```
Comment (gotjosh) [inline pkg/services/ngalert/notifier/external_alertmanager.go:284]: I think we already have an issue for this, could you link it on the PR? 🙏

**Review by gotjosh — APPROVED** [review]
LGTM

**Comment by gotjosh** [comment]
(make sure the linter is fixed before merging 😄)

File: pkg/services/ngalert/notifier/external_alertmanager.go (hunk 1 above)
Comment (santihernandezc) [reply pkg/services/ngalert/notifier/external_alertmanager.go:284]: It's not a public one :(

**Review by grobinson-grafana — APPROVED** [review]

**Pull request merged**